From 58633bd1712ff5c0bcedf519b09555376a779699 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:22:25 -0400
Subject: [cpufreq] acpi-cpufreq: use cpufreq_table_validate_and_show()

Message-id: <1409235879-1283-52-git-send-email-prarit@redhat.com>
Patchwork-id: 89388
O-Subject: [RHEL7.1 PATCH BZ 1134639 051/185] cpufreq: acpi-cpufreq: use cpufreq_table_validate_and_show()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 776b57be6581528125a076cbae5c0b75c50909f5
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Mon Sep 16 18:56:07 2013 +0530

    cpufreq: acpi-cpufreq: use cpufreq_table_validate_and_show()

    Lets use cpufreq_table_validate_and_show() instead of calling
    cpufreq_frequency_table_cpuinfo() and cpufreq_frequency_table_get_attr().

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/acpi-cpufreq.c b/drivers/cpufreq/acpi-cpufreq.c
index 8b0c7af..b93a144 100644
--- a/drivers/cpufreq/acpi-cpufreq.c
+++ b/drivers/cpufreq/acpi-cpufreq.c
@@ -837,7 +837,7 @@ static int acpi_cpufreq_cpu_init(struct cpufreq_policy *policy)
  data->freq_table[valid_states].frequency = CPUFREQ_TABLE_END;
  perf->state = 0;
 
- result = cpufreq_frequency_table_cpuinfo(policy, data->freq_table);
+ result = cpufreq_table_validate_and_show(policy, data->freq_table);
  if (result)
   goto err_freqfree;
 
@@ -868,8 +868,6 @@ static int acpi_cpufreq_cpu_init(struct cpufreq_policy *policy)
    (u32) perf->states[i].power,
    (u32) perf->states[i].transition_latency);
 
- cpufreq_frequency_table_get_attr(data->freq_table, policy->cpu);
-
  /*
   * the first call to ->target() should result in us actually
   * writing something to the appropriate registers.
-- 
1.7.1